service/options.go

```diff
@@ -4,11 +4,15 @@ import (
 	"flag"
 	"fmt"
 	log "github.com/Sirupsen/logrus"
+	"github.com/coreos/etcd/pkg/flags"
+	"os"
 	"strings"
 	"time"
 )
 
 type Options struct {
+	*flag.FlagSet
```

Comment:
It does not need to be embedded in options. `fs` should be just a local variable of `ParseRunOptions`.